File: HelpSoftware/Help/Help.Web/Web/src/app/EF/Booklet.ts
import { User } from "./EF";

export class Booklet {
  ID_Booklet: Number;
  Name: String;
  Description: string;
  User: User;
  Created: Date;
  LastModified: Date;
  Public: boolean;

  deserialize(input): Booklet {
    this.ID_Booklet = input.ID_Booklet;
    this.Name = input.Name;
    this.Description = input.Description;
    this.User = new User().deserialize(input.User);
    this.Created = input.Created;
    this.LastModified = input.LastModified;
    this.Public = input.Public;
    return this;
  }
}

File: HelpSoftware/Help/Help.EF/Entitys/Booklet.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Help.EF
{
  public class Booklet : IHelpEntity
  {
    [Key]
    public virtual int ID_Booklet { get; set; }

    public virtual string Name { get; set; }

    public virtual string Description { get; set; }

    public virtual User User { get; set; }

    public virtual DateTime Created { get; set; }

    public virtual DateTime? LastModified { get; set; }

    public virtual bool Public { get; set; }

    [Timestamp]
    public virtual byte[] TimeStamp { get; set; }

    public string GetEntityName()
    {
      return "Booklet";
    }

    public int GetID()
    {
      return ID_Booklet;
    }

    public bool GetWriteHistoryEntry()
    {
      return true;
    }
  }
}